### DISCUSSION TITLE
Multiple issue while interacting with EntryPoint contract (Account Abstraction)

### DISCUSSION THREAD

**Discussion by shikhar229169** [Cyfrin/foundry-full-course-cu#2024] [Q&A]
I have the following script similar to what was taught in the account abstraction video but with some minor changes:
[https://gist.github.com/shikhar229169/5d63eac1de91bf3091e3f9e3907f3b70](https://gist.github.com/shikhar229169/5d63eac1de91bf3091e3f9e3907f3b70)

The script works totally fine with anvil.
But when I am running it on any other chain either mainnet or testnet, the `getUserOpHash` function fails on EntryPoint contract, just saying reverted.

Then, I thought of deploying my own EntryPoint contract, so I just replaced the `config.entryPoint` with my newly deployed EntryPoint contract (just to mention that I deployed the EntryPoint within the broadcast cheatcode).

At this time, the `getUserOpHash` worked totally fine but the `handleOps` function reverted with some error. And when I ran it with simulation the whole script showed success, means the `handleOps` was working perfectly during simulation but during actual txn it reverted.

**Comment by EngrPips** [comment]
I haven't taken this course yet, and I don't have any context on what is going on, but let's look forward to @ciaranightingale  or @PatrickAlphaC  response to this as soon as they are available.

**Comment by PatrickAlphaC** [comment]
I'd need more information to triage. Could you send:
1. The exact error your getting running each script.
2. The exact command you're using to run each script (with sensitive information redacted of course)
3. Your config

It might be a good idea to add `-vvvv` to your foundry script as well, this will give you more information on what's going on with your call. If `getUserOpHash` is failing, I'd be inclined to think you're calling the wrong address.

**Comment by mustby** [comment]
Hey @shikhar229169! Did you figure this out? I'm also getting "handleOps" Reverting when I run:

forge test --match-test testEntryPointCanExecuteCommands -vvvv

Let me know!

**Reply by EngrPips** [reply]
Hello @mustby, Can you provide the details requested by @PatrickAlphaC  above

**Reply by mustby** [reply]
Upon further review, I will start another thread since my issue is slightly different @EngrPips

**Reply by mustby** [reply]
Here is the new thread @EngrPips! Thank you for any and all help!!!! 

https://github.com/Cyfrin/foundry-full-course-cu/discussions/3194

**Reply by EngrPips** [reply]
> Here is the new thread @EngrPips! Thank you for any and all help!!!!
> 
> #3194

Thanks for sharing buddy.
